### PR TITLE
Nano: fix trace custom model which doesn't support `compute_output_shape`

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

Fix trace custom models which do not support `compute_output_shape`.
For models which do not support the `compute_output_shape` method, we now fake a tensor from input spec to do a inference(except for onnx, onnx does not need this inference), so it can be traced by openvino, and can be quantized by inc.
However, these models will not support `evaluate` method.

### CODE REVIEW

File: python/nano/src/bigdl/nano/deps/onnxruntime/tensorflow/tensorflow_onnxruntime_model.py [hunk 1]
```diff
@@ -53,17 +53,10 @@ def __init__(self, model, input_spec=None, onnxruntime_session_options=None,
...
-                elif input_spec is not None:
-                    input_shape = tensor_spec_to_shape(input_spec)
-                    self._output_shape = model.compute_output_shape(input_shape)
-                else:
-                    invalidInputError(False,
-                                      "Subclassed model must specify `input_spec` parameter.")
+                self._output_shape = try_compute_output_shape(model, input_spec,
+                                                              try_fake_inference=False)
+                if input_spec is None and hasattr(model, "input_shape"):
+                    input_spec = input_spec = tf.TensorSpec(model.input_shape, model.dtype)
```
Comment (Oscilloscope98) [inline python/nano/src/bigdl/nano/deps/onnxruntime/tensorflow/tensorflow_onnxruntime_model.py:59]: duplicated `input_spec` here?

**Review by Oscilloscope98 — APPROVED** [review]
Others LGTM

File: python/nano/src/bigdl/nano/deps/onnxruntime/tensorflow/tensorflow_onnxruntime_model.py (hunk 1 above)
Comment (MeouSker77) [reply python/nano/src/bigdl/nano/deps/onnxruntime/tensorflow/tensorflow_onnxruntime_model.py]: fixed

**Review by rnwang04 — APPROVED** [review]
LGTM

**Pull request merged**